src/types/store.zig

```diff
@@ -206,8 +211,14 @@ pub const Store = struct {
         self.descs.set(resolved.desc_idx, desc);
     }
 
-    /// Set a type variable to redirect to the provided redirect
-    pub fn setVarRedirect(self: *Self, target_var: Var, redirect_to: Var) Allocator.Error!void {
+    /// Set a type variable to redirect to the provided variables.
+    /// During type-checking, you probably don't want to use this function.
+    ///
+    /// IMPORTANT: When using this function during type checking, it's possible
+    /// to loose `rank` information! You should prefer to use regular `unify`
+    /// over this function, which correctly propagates rank, unless you already
+    /// know the two vars are of the same rank.
```

Comment:
This all makes sense! If that's the assumption (ranks are the same), could we make a debug-only assertion to verify that?